src/libraries/Common/src/System/Globalization/FormatProvider.Number.cs

```diff
@@ -515,17 +515,26 @@ private static unsafe bool ParseNumber(ref char* str, char* strEnd, NumberStyles
                             int exp = 0;
                             do
                             {
-                                exp = exp * 10 + (ch - '0');
-                                ch = ++p < strEnd ? *p : '\0';
-                                if (exp > 1000)
+                                // Check if we are about to overflow past our limit of 9 digits
+                                if (exp >= 100_000_000)
                                 {
-                                    exp = 9999;
+                                    // Set exp to Int.MaxValue to signify the requested exponent is too large. This will lead to an OverflowException later.
+                                    exp = int.MaxValue;
+                                    number.scale = 0;
```

Comment:
@tannergooding negative exponents already had another in-use code path, so we have to use `int.MaxValue` as our Overflow signifier instead of `-1`. Additonally, `number.scale` has to be set to `0` because sometimes it's non-zero at this point and the addition on line 542 will overflow.